scripts/populate_new_and_noteworthy_projects.py

```diff
@@ -0,0 +1,137 @@
+"""
+This will update node links on POPULAR_LINKS_NODE and NEW_AND_NOTEWORTHY_LINKS_NODE.
+"""
+import sys
+import logging
+import datetime
+import dateutil
+import operator
+import requests
+from modularodm import Q
+from website.app import init_app
+from website import models
+from framework.auth.core import Auth
+from scripts import utils as script_utils
+from framework.transactions.context import TokuTransaction
+from website.project.model import Pointer
+from website.settings import POPULAR_LINKS_NODE, NEW_AND_NOTEWORTHY_LINKS_NODE, QA_USER_IDS, URL_BASE as BASE
+
+logger = logging.getLogger(__name__)
+
+def retrieve_data(url):
+    """ Fetch data and decode json """
+    response = requests.get(url)
+    data = response.json()
+    return data
+
+def get_api_base_path():
+    """ Returns api base path depending on environment"""
+    if BASE == 'PRODUCTION':
+        base = 'https://osf.io/'
+    elif BASE == 'STAGING':
+        base = 'https://staging.osf.io/'
+    elif BASE == 'STAGING2':
+        base = 'https://staging2.osf.io/'
+    else:
+        base = 'http://127.0.0.1:5000/'
+    return base
+
+def get_apiv2_base_path():
+    """ Returns apiv2 base path depending on environment"""
+    if BASE == 'PRODUCTION':
+        base = 'https://api.osf.io/'
+    elif BASE == 'STAGING':
+        base = 'https://staging-api.osf.io/'
+    elif BASE == 'STAGING2':
+        base = 'https://staging2-api.osf.io/'
+    else:
+        base = 'http://127.0.0.1:8000/'
+    return base
+
+def get_popular_nodes():
+    """ Fetch data from url that returns dict with a list of popular nodes from piwik """
+    discover_url = get_api_base_path() + 'api/v1/explore/activity/popular/raw/'
+    return retrieve_data(discover_url)
+
+def get_new_and_noteworthy_nodes():
+    """ Fetches nodes created in the last month and returns 25 sorted by highest log activity """
+    today = datetime.datetime.now()
+    last_month = (today - dateutil.relativedelta.relativedelta(months=1)).isoformat()
+    discover_url = get_apiv2_base_path() + \
+                   'v2/nodes/?sort=-date_created&page[size]=1000&related_counts=True&filter[date_created][gt]={}'.format(last_month)
+    data = retrieve_data(discover_url)['data']
+    node_log_count_mapping = {}
+    for new_node in data:
+        node_log_count_mapping[new_node['id']] = new_node['relationships']['logs']['links']['related']['meta']['count']
+    sort_by_log_count = sorted(node_log_count_mapping.items(), key=operator.itemgetter(1), reverse=True)
+    sorted_node_ids = [node[0] for node in sort_by_log_count]
+    return sorted_node_ids[:25]
+
+def is_eligible_node(node):
+    """
+    Check to ensure that node is not the POPULAR or NEW_AND_NOTEWORTHY LINKS_NODE.
+    Ensures QA members are not contributors
+    """
+    if node._id == POPULAR_LINKS_NODE or node._id == NEW_AND_NOTEWORTHY_LINKS_NODE:
+        return False
+
+    for contrib in node.contributors:
+        if contrib._id in QA_USER_IDS:
```

Comment:
We might want to use a more generic name for this setting; we might not only exclude QA users. How about something like `NEW_AND_NOTEWORTHY_CONTRIBUTOR_BLACKLIST`?